packages/editor-ui/src/components/ValueSurvey.vue

```diff
@@ -0,0 +1,274 @@
+						<n8n-input
+							v-model="form.email"
+							placeholder="Your email address"
+							size="medium"
+							@input="onInputChange"
+						/>
+						<div :class="$style.button">
+							<n8n-button label="Send" float="right" @click="send" :disabled="!isEmailValid" />
+						</div>
+					</div>
+					<div :class="$style.disclaimer">
+						<n8n-text size="small" color="text-xlight">
+							David from our product team will get in touch personally
+						</n8n-text>
+					</div>
+				</div>
+			</section>
+		</template>
+	</ModalDrawer>
+</template>
+
+<script lang="ts">
+import { VALID_EMAIL_REGEX, VALUE_SURVEY_MODAL_KEY } from '@/constants';
+import { IN8nPromptResponse } from '@/Interface';
+
+import ModalDrawer from './ModalDrawer.vue';
+
+import mixins from 'vue-typed-mixins';
+import { workflowHelpers } from '@/components/mixins/workflowHelpers';
+
+const DEFAULT_TITLE = `How likely are you to recommend n8n to a friend or colleague?`;
+const GREAT_FEEDBACK_TITLE = `Great to hear! Can we reach out to see how we can make n8n even better for you?`;
+const DEFAULT_FEEDBACK_TITLE = `Thanks for your feedback! We'd love to understand how we can improve. Can we reach out?`;
+
+export default mixins(workflowHelpers).extend({
+	name: 'ValueSurvey',
+	components: {
+		ModalDrawer,
+	},
+	computed: {
+		getTitle(): string {
+			if (this.form.value !== '') {
+				if (Number(this.form.value) > 7) {
+					return GREAT_FEEDBACK_TITLE;
+				} else {
+					return DEFAULT_FEEDBACK_TITLE;
+				}
+			} else {
+				return DEFAULT_TITLE;
+			}
+		},
+		isEmailValid(): boolean {
+			return VALID_EMAIL_REGEX.test(String(this.form.email).toLowerCase());
+		},
+	},
+	data() {
+		return {
+			form: {
+				email: '',
+				value: '',
+			},
+			showButtons: true,
+			VALUE_SURVEY_MODAL_KEY,
+		};
+	},
+	methods: {
+		closeDialog(): void {
+			if (this.form.value === '') {
+				this.$telemetry.track('User responded value survey score', {
+					instance_id: this.$store.getters.instanceId,
+					nps: '',
+				});
+			} else {
+				this.$telemetry.track('User responded value survey email', {
+					instance_id: this.$store.getters.instanceId,
+					email: '',
+				});
+			}
+
+			this.$store.commit('ui/closeTopModal');
+		},
+		onInputChange(value: string) {
+			this.form.email = value;
+		},
+		selectSurveyValue(value: string) {
+			this.form.value = value;
+			this.showButtons = false;
+			this.$store
+				.dispatch('settings/submitValueSurvey', { value: this.form.value })
+				.then((response: IN8nPromptResponse) => {
+					if (response.updated) {
+						this.$telemetry.track('User responded value survey score', {
+							instance_id: this.$store.getters.instanceId,
+							nps: this.form.value,
+						});
+					}
+				});
+		},
+		send(): void {
+			if (this.isEmailValid) {
+				this.$store
+					.dispatch('settings/submitValueSurvey', {
+						email: this.form.email,
+						value: this.form.value,
+					})
+					.then((response: IN8nPromptResponse) => {
```

Comment:
async/await is always preferable to promises

Reply:
I changed to async/await everywhere I found in ValueSurvey and ContactPrompt modals.